airflow/providers/amazon/aws/hooks/s3.py

```diff
@@ -198,8 +205,12 @@ def check_for_prefix(self, prefix, delimiter, bucket_name=None):
         return False if plist is None else prefix in plist
 
     @provide_bucket_name
-    def list_prefixes(self, bucket_name=None, prefix='', delimiter='',
-                      page_size=None, max_items=None):
+    def list_prefixes(self,
+                      bucket_name: Optional[str] = None,
+                      prefix: str = '',
+                      delimiter: str = '',
+                      page_size: Optional[int] = None,
+                      max_items: Optional[int] = None) -> Optional[list]:
```

Comment:
```suggestion
                      prefix: Optional[str] = None,
                      delimiter: Optional[str] = None,
                      page_size: Optional[int] = None,
                      max_items: Optional[int] = None) -> Optional[list]:
           prefix = prefix or ""
           delimiter = delimiter or ""
```
WDYT?

Reply:
I think we should avoid strings as default values

Reply:
I think it makes more sense if the idea behind `Optional[str]` is that it can be a `str` or `None`, yep.